### PR TITLE
Create openocd CMakelists

### PR DESCRIPTION
**Pull request by guanyung-rs**
> ### Motivate of the pull request
> - [ ] To address an existing issue. If so, please provide a link to the issue: <issue id>
> - [x] Breaking new feature. If so, please describe details in the description part.

> ### Describe the technical details

This PR add openocd_cmakelists into FOEDAG. This changelist is similar to gtkwave_cmake where
- openocd executable is downloaded from RS fork version of openocd repo.
- The openocd executable is then copy to the bin folder
- It also copy the gemini.cfg into share/foedag/configuration folder which is used by the program_device tcl command call

testing
- tested local built `make` and confirm openocd is copied to bin directory, gemini.cfg is copied to share/foedag/configuration
- tested `make install PREFIX="home-test-directory"`. The files are installed in the expected location

Some screens shots
Build directory

Install directory

> #### What is currently done? (Provide issue link if applicable)
> 
> 
> 
> 
> 
>
> #### What does this pull request change?
> 
> 
> 
> 
> 

> ### Which part of the code base require a change
> 
> - [ ] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [ ] Require a change on Quality of Results (QoR)
> - [ ] Break back-compatibility. If so, please list who may be influenced.

### CODE REVIEW

File: third_party/openocd_cmake/CMakeLists.txt
Comment (nadeemyaseen-rs) [inline third_party/openocd_cmake/CMakeLists.txt:72]: Are we using it somewhere?

File: third_party/openocd_cmake/CMakeLists.txt
Comment (nadeemyaseen-rs) [inline third_party/openocd_cmake/CMakeLists.txt:54]: Is this needed here? It is already set at line number 30.

File: third_party/openocd_cmake/CMakeLists.txt
Comment (guanyung-rs) [reply third_party/openocd_cmake/CMakeLists.txt:72]: ops. I think line72 should be deleted. I will clean up the code.

File: third_party/openocd_cmake/CMakeLists.txt
Comment (guanyung-rs) [reply third_party/openocd_cmake/CMakeLists.txt:54]: I will remove it.

**Comment by guanyung-rs** [comment]
There is something I need to fix. Please don't merge the PR.

**Comment by guanyung-rs** [comment]
> There is something I need to fix. Please don't merge the PR.

I fixed the openocd download as there was changes in the openocd build yesterday. I also posted all the top level FOEDAG_RS and Raptor PR

File: third_party/openocd_cmake/CMakeLists.txt
Comment (ecchg) [inline third_party/openocd_cmake/CMakeLists.txt:47]: As we talked about, it's possible to merge this after line 25. However, this is being done to improve the sequence of information for better readability.

**Review by ecchg — APPROVED** [review]

**Pull request merged**
